ddtrace/llmobs/decorators.py

```diff
@@ -66,8 +67,15 @@ def wrapper(*args, **kwargs):
                 if span_name is None:
                     span_name = func.__name__
                 traced_operation = getattr(LLMObs, operation_kind, "workflow")
-                with traced_operation(name=span_name, session_id=session_id, ml_app=ml_app):
-                    return func(*args, **kwargs)
+                with traced_operation(name=span_name, session_id=session_id, ml_app=ml_app) as span:
+                    func_signature = signature(func)
+                    bound_args = func_signature.bind_partial(*args, **kwargs)
+                    if _automatic_io_annotation and bound_args.arguments:
+                        LLMObs.annotate(span=span, input_data=bound_args.arguments)
+                    resp = func(*args, **kwargs)
+                    if _automatic_io_annotation and resp and operation_kind != "retrieval":
+                        LLMObs.annotate(span=span, output_data=resp)
```

Comment:
if a user manually annotated `output_data` I think it will be over-written here? Or would a user need to specify `_automatic_io_annotation=False` every time they want to manually annotate a span? 

I'm guessing we need to add logic in annotate to make sure automatic annotations do not override manual annotations

Reply:
Discussed offline - we will only do auto-annotations if manual annotations are not present